From 02037d858887ba24978cc71b55277d67fef18c3c Mon Sep 17 00:00:00 2001
From: David Clark <dclark@sierrawireless.com>
Date: Wed, 17 Jan 2018 08:47:44 -0800
Subject: [PATCH 1/2] Add API to retrieve AP BSSID

Added BSSID size to the API definitions and added new AP get function.

Resolves: LE-9186
Change-Id: Id90c3a65d64f5da2f840f5d9c578f8b7ab445c04
---
 interfaces/wifi/le_wifiClient.api | 18 +++++++++++++++++-
 interfaces/wifi/le_wifiDefs.api   | 15 +++++++++++++++
 2 files changed, 32 insertions(+), 1 deletion(-)

diff --git a/interfaces/wifi/le_wifiClient.api b/interfaces/wifi/le_wifiClient.api
index b9ef96cbf..4bcfafa5a 100644
--- a/interfaces/wifi/le_wifiClient.api
+++ b/interfaces/wifi/le_wifiClient.api
@@ -339,6 +339,22 @@ FUNCTION int16 GetSignalStrength
 
 //--------------------------------------------------------------------------------------------------
 /**
+ * Get the Basic Service set identifier (BSSID) of the AccessPoint
+ *
+ * @return
+ *      - LE_FAULT if the function failed.
+ *      - LE_BAD_PARAMETER if some parameter is invalid.
+ *      - LE_OK if the function succeeded.
+ */
+//--------------------------------------------------------------------------------------------------
+FUNCTION le_result_t GetBssid
+(
+    AccessPoint accessPointRef IN,  ///< WiFi Access Point reference.
+    string bssid[le_wifiDefs.MAX_BSSID_BYTES] OUT ///< The BSSID
+);
+
+//--------------------------------------------------------------------------------------------------
+/**
  * Get the Service set identification (SSID) of the AccessPoint
  * @note that the SSID does not have to be human readable ASCII values, but often has.
  *
@@ -351,7 +367,7 @@ FUNCTION int16 GetSignalStrength
 FUNCTION le_result_t GetSsid
 (
     AccessPoint accessPointRef IN,  ///< WiFi Access Point reference.
-    uint8 Ssid[le_wifiDefs.MAX_SSID_BYTES] OUT ///< The SSID returned as a octet array.
+    uint8 ssid[le_wifiDefs.MAX_SSID_BYTES] OUT ///< The SSID returned as a octet array.
 );
 
 
diff --git a/interfaces/wifi/le_wifiDefs.api b/interfaces/wifi/le_wifiDefs.api
index 57481e955..bd5b986e0 100644
--- a/interfaces/wifi/le_wifiDefs.api
+++ b/interfaces/wifi/le_wifiDefs.api
@@ -135,3 +135,18 @@ DEFINE MAX_SSID_LENGTH       = 32;
 //--------------------------------------------------------------------------------------------------
 DEFINE MAX_SSID_BYTES        = (MAX_SSID_LENGTH + 1);
 
+//--------------------------------------------------------------------------------------------------
+/**
+ * The maximum length of octets of the Basic Service set identifier (BSSID). The length is derived
+ * from 6 bytes represented as hexadecimal character string with bytes separated by colons.
+ */
+//--------------------------------------------------------------------------------------------------
+DEFINE MAX_BSSID_LENGTH      = 17;
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Maximum length of the Basic Service set identifier (BSSID).
+ * One extra byte is added for the null character.
+ */
+//--------------------------------------------------------------------------------------------------
+DEFINE MAX_BSSID_BYTES       = (MAX_BSSID_LENGTH + 1);
-- 
2.11.0

